Задание 2.
Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

In [46]:
from math import factorial as fac

Создадими функцию для расчета сочетания.

In [47]:
def combinations(n, k):
    return fac(n) / (fac(k) * fac(n - k))

Запишем исходные условия в переменные.

In [48]:
n = 5000
k = 2
p = 0.0004
q = 1 - p

Применим формулу Бернулли $$P_n(X=k) = C_n^k p^k q^{n-k}$$

In [49]:
P = combinations(n, k) * (p**k) * (q**(n-k))
P

0.2707247150266753

Получаем 0.27 или 27%, что в первый день перегорят 2 лампочки.

Для второй части задачи переопределим переменные.

In [50]:
n = 5000
k = 0
p = 0.0004
q = 1 - p

Так же по формуле Бернулли считаем.

In [51]:
P = combinations(n, k) * (p**k) * (q**(n-k))
P

0.13528114551440706

Получаем 0.135 или 13.5%, что в первый день НЕ перегорит ни одна лампочка.